tools/refguide_check.py

```diff
@@ -595,23 +611,25 @@ def check_output(self, want, got, optionflags):
                 s_got = ", ".join(s_got[1:-1].split())
                 return self.check_output(s_want, s_got, optionflags)
 
+            if "=" not in want and "=" not in got:
+                # if we're here, want and got cannot be eval-ed (hence cannot
+                # be converted to numpy objects), they are not namedtuples
+                # (those must have at least one '=' sign).
+                # Thus they should have compared equal with vanilla doctest.
+                # Since they did not, it's an error.
+                return False
+
             if not self.parse_namedtuples:
                 return False
             # suppose that "want"  is a tuple, and "got" is smth like
             # MoodResult(statistic=10, pvalue=0.1).
             # Then convert the latter to the tuple (10, 0.1),
             # and then compare the tuples.
             try:
-                num = len(a_want)
-                regex = (r'[\w\d_]+\(' +
-                         ', '.join([r'[\w\d_]+=(.+)']*num) +
-                         r'\)')
-                grp = re.findall(regex, got.replace('\n', ' '))
-                if len(grp) > 1:  # no more than one for now
-                    return False
-                # fold it back to a tuple
-                got_again = '(' + ', '.join(grp[0]) + ')'
-                return self.check_output(want, got_again, optionflags)
+
+                got_again = try_convert_namedtuple(got)
+                want_again = try_convert_namedtuple(want)
+                return self.check_output(want_again, got_again, optionflags)
```

Comment:
Can we move this part to an `else` clause of `try/except`? Ideally we'd only have code in the `try/except` that we expect to fail for a good reason. I assume the "good reason" here is that it couldn't be converted to namedtuple, rather than `check_output` throwing some error.

Reply:
Readability is in the eyes of the reviewer, so done --- moved the return line out of the except part. Is that what you meant Eric?